scipy/stats/_stats_py.py

```diff
@@ -1013,6 +1013,14 @@ def _moment(a, moment, axis, *, mean=None):
         # Starting point for exponentiation by squares
         mean = a.mean(axis, keepdims=True) if mean is None else mean
         a_zero_mean = a - mean
+
+        eps = np.finfo(a_zero_mean.dtype).resolution * 10
+        rel_diff = np.max(np.abs(a_zero_mean), axis=axis, keepdims=True)/mean
+        if np.any((0 < rel_diff) & (rel_diff < eps)):
```

Comment:
We could remove the `(0 < rel_diff)` but there are some tests with scalar input that start failing, and I wasn't sure whether we wanted to just warn or also return nan.